### PR TITLE
Prepare for publishing

### PR DESCRIPTION
**Pull request by ucbjrl**
Introduce Scala/Java version dependencies.
Introduce defaultVersions stanza.
Bump ScalaTest version.
Bump riscv-mini version (to align with other 3.3 releases)
Add Maven publishing definitions.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -13,8 +50,51 @@ val commonSettings = Seq(
...
+</scm>
+<developers>
+  <developer>
+    <id>donggyukim</id>
+    <name>Donggyu Kim</name>
+    <url>http://donggyukim.github.io/</url>
+  </developer>
+  <developer>
+    <id>chick</id>
+    <name>Charles Markley</name>
```
Comment (chick) [inline build.sbt]: NBD but can I go with Chick here

File: build.sbt [hunk 2]
```diff
@@ -1,10 +1,47 @@
...
+
+// Provide a managed dependency on X if -DXVersion="" is supplied on the command line.
+val defaultVersions = Seq(
+  "chisel3" -> "3.3-SNAPSHOT"
+)
+
 val commonSettings = Seq(
   scalaVersion := "2.12.6",
-  crossScalaVersions := Seq("2.11.12", "2.12.4"),
+  crossScalaVersions := Seq("2.12.6", "2.11.12"),
```
Comment (chick) [inline build.sbt]: I notice that Chisel3 is 2.12.10, should that be the value here?

File: build.sbt (hunk 1 above)
Comment (ucbjrl) [reply build.sbt]: Will do.

File: build.sbt (hunk 2 above)
Comment (ucbjrl) [reply build.sbt]: Good catch.

Comment (ucbjrl) [reply build.sbt]: I'll also bump the `sbt` version to 1.3.2

**Pull request merged**